Change Direct Message warning to make admins' ability to access them clearer

ClearlyClaire opened the pull request:
Fixes #18079

The current English warning is apparently not enough to convey that server administrators are technically able to access Direct Messages, as users are still surprised after learning about it outside of Mastodon.

The message introduced in #7420 was more explicit about it, but was later changed in #7546. The concern at that point seems to have been that it may alarm users more than necessary.

However, the French translation remained as a loose translation of the message introduced in #7420, and I have never witnessed anyone being upset at this warning, so I think it is actually fine.

ClearlyClaire committed:
Change Direct Message warning to make admins' ability to access them clearer

Fixes #18079

shleeable commented:
Please append this image. 
![image](https://user-images.githubusercontent.com/17537000/165426982-983aa588-3089-4b6d-bd48-01d31c4bd87f.png)

n0toose commented:
@shleeable Agreed. But don't other applications (e.g. Admin FE of Pleroma, I remember using that like two years ago and going all like o_O) provide the opportunity to read such messages *very, very* easily?

It's just that, fun fact, I remember messaging this friend of mine in a Minecraft server and this moderator comes out of nowhere to ask me to stop cussing in the private messages that he was reading, because he was reading them in real-time and he did not want to read cuss words. Some horrible communities are managed by horrible people, some of them willing to put the extra effort as some sort of a weird power trip, and that also affects people that are not a part of them as well here. :V

shleeable commented:
> It's just that, fun fact, I remember messaging this friend of mine in a Minecraft server and this moderator comes out of nowhere to ask me to stop cussing in the private messages that he was reading, because he was reading them in real-time and he did not want to read cuss words. Some horrible communities are managed by horrible people, some of them willing to put the extra effort as some sort of a weird power trip, and that also affects people that are not a part of them as well here. :V

That sucks... It does make sense to assume we're not all good actors. Just because I am, doesn't mean they're not.

n0toose commented:
> That sucks...

Don't worry, I definitely kept swearing. I think that it would make people go like "Oh no, how can I message people in private then?". I also work on privacy-related projects and therefore believe that it's very important for people to know the fine-prints to their privacy, as that will ultimately protect them better, make them skeptical and also develop a sense of trust in a platform that aims to take decisions in the best interest of its users.

I just took a look at [the documentation, which seems to be very direct](https://docs.joinmastodon.org/user/posting/#direct) about this, more than a privacy policy (even `mastodon.social`'s) does. I haven't noticed any other instances of an instance linking back to that "centralized" source of information, however, so I am not sure if that's an intentional decision or not.

Brawaru commented:
> It's just that, fun fact, I remember messaging this friend of mine in a Minecraft server and this moderator comes out of nowhere

they must've saw the console maybe, because darn minecraft logs every command you execute and thus direct messages...

n0toose commented:
@Brawaru They were a moderator and there's a specific permission in the Essentials plugin that you have to enable which allows you to snoop on everyone's conversations. Knowing the circumstances and having hosted both Fedi and Minecraft servers, I think that the situations are comparable.

n0toose commented:
While we're at it, is it also not important to mention that if you mention someone else (e.g. warning others of harassment), they will be able to see the conversation?

I saw someone on Mastodon complaining about how the UI/UX doesn't warn you about that and that "people keep falling for it" and was like "huh, I could see that happening".

tribela commented:
Agree to shleeable.
I don't want to scare/worry people pre-cause.
And "admin can access your data (if they decided to do that shxt)" is pretty common sense as i think.

ClearlyClaire commented:
This has been superseded by #18146 which:
- changed the DM icon from an envelope to a `@`
- renamed the “Direct” visibility option to “Only people I mention”
- removed the direct option from profile and status dropdown menus
- changed the warning to “Posts on Mastodon are not end-to-end encrypted. Do not share any dangerous information over Mastodon.”
- changed the title of the DM column to “Conversations”
